Add support for ROG STRIX X870-I GAMING WIFI

mzonski opened the pull request:
In this pull request, I am adding support for monitoring temperatures on the [ASUS ROG STRIX X870-I GAMING WIFI](https://rog.asus.com/motherboards/rog-strix/rog-strix-x870-i-gaming-wifi/).

I have been using and testing this implementation for a while now, but finally decided to upstream my changes.

```
asusec-isa-0000
Adapter: ISA adapter
CPU:          +56.0°C  
CPU Package:  +60.0°C  
Motherboard:  +44.0°C  
VRM:          +52.0°C  
```
[I have ported](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/1794) support for AMD 800 series to `LibreHardwareMonitor`, where I directly compared values between the proprietary implementation and the changes you can find in my commit. The values matched. I tried to find more values by reading the EC controller, but for now, I am still learning, and failed to do so.

zeule commented:
Thanks for the contribution! Could you, please, share DSDT dump? Have you tried to find a specific mutex?

zeule commented:
Also please sing-off the commit (`git commit -s`) so that it can be submitted to the mainline kernel.

mzonski commented:
Hello, first of all, sorry for not noticing the contribution guide at the first place.

Here you can find two files you ask to include:
[DSDT](https://drive.google.com/file/d/1rU6bMpSwPx8-KcKsAKsH4L8hpnUjBt6y/view)
[dmidecode output](https://drive.google.com/file/d/1bNWuxPYsmhqWUQSTw_ThFa_--Tr1Iiyr/view)

I think I am signing my commits with PGP by default, it says "Verified" next to the commit I am pushin in Github, also my Git client says the commits are signed.

I gave up trying to find anything new, because bruteforcing my way didn't really got me anywhere, and that is how I was trying to do on Windows when developing LibreHardwareMonitor patch.

If you don't mind I would like to spend some time on DSDT analysis, as finally I have the generic overview what I should achieve.

zeule commented:
Hmm, https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/1794 list it for the EC.

mzonski commented:
I have tested usage of `_SB_.PCI0.SBRG.SIO1.MUT0` mutex. The sensors are still reading the correct data, as they worked with previous. I am getting the kernel warnings as you pointed out in previous comment, during querying 2nd EC bank with my custom application written specifically to check/dump values of EC controller.

```
Sep 06 15:41:40 corn kernel: hwmon hwmon4: Concurrent access to the ACPI EC detected.
                             Race condition possible.
Sep 06 15:41:42 corn kernel: hwmon hwmon4: Concurrent access to the ACPI EC detected.
                             Race condition possible.
Sep 06 15:41:43 corn kernel: hwmon hwmon4: Concurrent access to the ACPI EC detected.
                             Race condition possible.
Sep 06 15:41:45 corn kernel: hwmon hwmon4: Concurrent access to the ACPI EC detected.
                             Race condition possible.
```


I have the nct6775 module installed in my kernel, it was the only way that I discovered to control fans from the system.
```
nct6799-isa-0290
Adapter: ISA adapter
in0:                          824.00 mV (min =  +0.00 V, max =  +1.74 V)
in1:                            1.02 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in2:                            3.39 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in3:                            3.41 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in4:                            1.01 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in5:                            1.03 V  (min =  +0.00 V, max =  +0.00 V)
in6:                          488.00 mV (min =  +0.00 V, max =  +0.00 V)  ALARM
in7:                            3.39 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in8:                            3.36 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in9:                            3.41 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in10:                           1.14 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in11:                           1.12 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in12:                           1.03 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in13:                         224.00 mV (min =  +0.00 V, max =  +0.00 V)  ALARM
in14:                           2.04 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in15:                         672.00 mV (min =  +0.00 V, max =  +0.00 V)  ALARM
in16:                           3.39 V  (min =  +0.00 V, max =  +0.00 V)  ALARM
in17:                         904.00 mV (min =  +0.00 V, max =  +0.00 V)
fan1:                          634 RPM  (min =    0 RPM)
fan2:                          820 RPM  (min =    0 RPM)
fan6:                         3515 RPM  (min =    0 RPM)
fan7:                         1284 RPM  (min =    0 RPM)
SYSTIN:                        +45.0°C  (high = +80.0°C, hyst = +75.0°C)
                                        (crit = +125.0°C)  sensor = thermistor
CPUTIN:                        +40.5°C  (high = +80.0°C, hyst = +75.0°C)
                                        (crit = +125.0°C)  sensor = thermistor
AUXTIN0:                       +59.0°C  (high = +80.0°C, hyst = +75.0°C)
                                        (crit = +127.0°C)  sensor = thermistor
AUXTIN1:                       +19.0°C  (high = +80.0°C, hyst = +75.0°C)
                                        (crit = +127.0°C)  sensor = thermistor
AUXTIN2:                       +19.0°C  (high = +80.0°C, hyst = +75.0°C)
                                        (crit = +127.0°C)  sensor = thermistor
AUXTIN3:                       -62.0°C  (high = +80.0°C, hyst = +75.0°C)
                                        (crit = +125.0°C)  sensor = thermistor
AUXTIN4:                       +24.0°C  (high = +80.0°C, hyst = +75.0°C)
                                        (crit = +125.0°C)
PECI/TSI Agent 0 Calibration:  +40.0°C  (high = +80.0°C, hyst = +75.0°C)
AUXTIN5:                       +31.0°C  
PCH_CHIP_CPU_MAX_TEMP:          +0.0°C  
PCH_CHIP_TEMP:                  +0.0°C  
PCH_CPU_TEMP:                   +0.0°C  
TSI0_TEMP:                     +40.1°C  
intrusion0:                   ALARM
intrusion1:                   ALARM
beep_enable:                  disabled

asusec-isa-0000
Adapter: ISA adapter
CPU:          +40.0°C  
CPU Package:  +40.0°C  
Motherboard:  +45.0°C  
VRM:          +46.0°C  

```

zeule commented:
Now I'm confused. The mutex we use in this driver is to protect from data race with the board firmware. If you use another application that does not use the same mutex, the data race is to be expected. 

I think T_Sensor is AUXTIN3 in nct6799-isa-0290.

mzonski commented:
I am attaching pictures I took from my BIOS.
It's reporting `T_Sensor` as unavailable, the `VRM_HS_FAN` is already populated out of the box with some connector, BIOS reports it's missing but it is really not. I haven't traced this connection, but it really doesn't matter now.

As you see, I guess this board could be facing some challenges. I am also attaching the image of the board reported by BIOS.

![20250906_162332](https://github.com/user-attachments/assets/c33acb27-9660-4de8-b182-1a4be0d59b58)
![20250906_162320](https://github.com/user-attachments/assets/0fc2474a-f20b-4df0-8b0b-67f99b8800b6)
![20250906_162308](https://github.com/user-attachments/assets/d84c8bd7-09af-4c22-b2f0-201311f693cb)
![20250906_162141](https://github.com/user-attachments/assets/2a29166b-091a-421b-8ca5-5c84829b6210)

Please let me know what do you think about keeping the `CPU_Opt` and `T_Sensor` value given the data I provided.

zeule commented:
There are many sensors in the EC registers we don't know how to read (yet), but that is of course outside of the scope of this PR. CPU_Opt is not available for this board, please remove that. T_Sensors is likely reported via the nct driver. Also please fix formatting (only tabs are allowed for indentation).

I still don't understand the situation with the mutex. Do the warnings appear when only the firmware and `sensors` are active?

mzonski commented:
The warnings don't appear at all when I am running this command:
```
for i in {1..10000}; do sensors asusec-isa-0000 > /dev/null; done
```

They also don't appear when I am trying to read first page of EC controller using my app, but once I try to read second page I am getting this errors. But I wouldn't say the implementation of my app is good enough, so I wouldn't bother investigating it.

mzonski commented:
The code was formatted using tabs, but double-checked it, and I think I got none whitespaces in the indentation. If you see any, please leave a remark and I'll fix it.

I have removed the `CPU_Opt`, but left the `T_Sensor`.

zeule commented:
Formatting is correct now, thank you! But I don't understand why did you retain the T_Sensor. For all the other boards this driver supports, disconnected T_Sensors results in negative reading (either -40 or -63), which is not the case here. But we see such value in the nct output. I conclude the T_Sensors is reported via the SIO chip registers. I don't understand how ASUS engineers allocate sensor to the EC and SIO. Looks like EC receives those which do not fit into the SIO register set. 

> They also don't appear when I am trying to read first page of EC controller using my app, but once I try to read second page I am getting the warnings.

That's because race is only detected when the driver finds EC bank set to anything else but 0, and it always leaves it set to 0, like the firmware does.

zeule commented:
If you don't want to make any more changes, I merge this, please.

mzonski commented:
I am content with making any changes, it is your repository and your rules, and if I want to contribute I will follow them, it's very simple.

I have modified my previous response, and removed `T_Sensor` from the code after reading for the second time your comment about system possibly getting this value from the `NCT6701D` chip.

If you don't think I need to do any more changes I will be happy if you merge it.

mzonski commented:
Nevertheless, thank you for supporting me during this pull request, your input was very valuable, especially regarding the mutex. I learned something new, and for me this matters the most.

zeule commented:
Let me thank you once again for the contribution, testing efforts and your time investments! The changeset was submitted to the mainline kernel and if accepted quickly will be released with version 6.18.
